### DISCUSSION TITLE
Can you "format" badge URLs for better look in raw view?

### DISCUSSION THREAD

**Discussion by Andre601** [badges/shields#9497] [Q&A] [answered]
This may be a rather unusual question and isn't quite about shields itself, but I hope to receive an answer either way.

So... I currently use a dynamic badge URL with a custom, base64 encoded SVG image. Given these points, the URL becomes quite long and hard to read when viewing it in the Raw markdown file.

As an example, this is one of the URLs I currently use in the README.md file using a reference link:
```markdown
[asl-stars-badge]: https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcodeberg.org%2Fapi%2Fv1%2Frepos%2FAndre601%2FAdvancedServerList&query=%24.stars_count&logo=data:image/svg%2Bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiBmaWxsPSIjZDRkYmRmIj48cGF0aCBkPSJtMTIuNjcyLjY2OCAzLjA1OSA2LjE5NyA2LjgzOC45OTNhLjc1Ljc1IDAgMCAxIC40MTYgMS4yOGwtNC45NDggNC44MjMgMS4xNjggNi44MTJhLjc1Ljc1IDAgMCAxLTEuMDg4Ljc5TDEyIDE4LjM0N2wtNi4xMTYgMy4yMTZhLjc1Ljc1IDAgMCAxLTEuMDg4LS43OTFsMS4xNjgtNi44MTEtNC45NDgtNC44MjNhLjc0OS43NDkgMCAwIDEgLjQxNi0xLjI3OWw2LjgzOC0uOTk0TDExLjMyNy42NjhhLjc1Ljc1IDAgMCAxIDEuMzQ1IDBaIj48L3BhdGg+PC9zdmc+&style=for-the-badge&label=&color=061825&cacheSeconds=86400
```

My question now is, if there is some way in markdown to format this URL a bit more nicely to make it easier to at least see the different query parameters used.

Like to give a clear example, I ask if I can have the URL formatted like this while still having it work:
```markdown
https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcodeberg.org%2Fapi%2Fv1%2Frepos%2FAndre601%2FAdvancedServerList
&query=%24.stars_count
&logo=data:image/svg%2Bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiBmaWxsPSIjZDRkYmRmIj48cGF0aCBkPSJtMTIuNjcyLjY2OCAzLjA1OSA2LjE5NyA2LjgzOC45OTNhLjc1Ljc1IDAgMCAxIC40MTYgMS4yOGwtNC45NDggNC44MjMgMS4xNjggNi44MTJhLjc1Ljc1IDAgMCAxLTEuMDg4Ljc5TDEyIDE4LjM0N2wtNi4xMTYgMy4yMTZhLjc1Ljc1IDAgMCAxLTEuMDg4LS43OTFsMS4xNjgtNi44MTEtNC45NDgtNC44MjNhLjc0OS43NDkgMCAwIDEgLjQxNi0xLjI3OWw2LjgzOC0uOTk0TDExLjMyNy42NjhhLjc1Ljc1IDAgMCAxIDEuMzQ1IDBaIj48L3BhdGg+PC9zdmc+
&style=for-the-badge
&label=
&color=061825
&cacheSeconds=86400
```

If there is no such way, then that's fine.

**Comment by chris48s** [comment, marked as answer]
In some markdown dialects, this works

```
![alt text](
    https://img.shields.io/badge/dynamic/json?url=https%3A%2F%2Fcodeberg.org%2Fapi%2Fv1%2Frepos%2FAndre601%2FAdvancedServerList
&query=%24.stars_count
&logo=data:image/svg%2Bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyNCAyNCIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiBmaWxsPSIjZDRkYmRmIj48cGF0aCBkPSJtMTIuNjcyLjY2OCAzLjA1OSA2LjE5NyA2LjgzOC45OTNhLjc1Ljc1IDAgMCAxIC40MTYgMS4yOGwtNC45NDggNC44MjMgMS4xNjggNi44MTJhLjc1Ljc1IDAgMCAxLTEuMDg4Ljc5TDEyIDE4LjM0N2wtNi4xMTYgMy4yMTZhLjc1Ljc1IDAgMCAxLTEuMDg4LS43OTFsMS4xNjgtNi44MTEtNC45NDgtNC44MjNhLjc0OS43NDkgMCAwIDEgLjQxNi0xLjI3OWw2LjgzOC0uOTk0TDExLjMyNy42NjhhLjc1Ljc1IDAgMCAxIDEuMzQ1IDBaIj48L3BhdGg+PC9zdmc+
&style=for-the-badge
&label=
&color=061825
&cacheSeconds=86400
)
```

but not in GitHub Flavoured Markdown. GitHub treats the linebreaks as whitespace.

I am not aware of a way to split A URL across multiple lines in GitHub Flavoured Markdown.

**Comment by calebcartwright** [comment]
I don't feel like I can answer authoritatively and unequivocally given the wide breadth of markdown flavors and support across different platforms, but I'd say no, I don't believe you can insert newline characters within markdown links without actually breaking the url (happy to be shown otherwise though, as I agree they'd be helpful!)

In the case of a url that has a custom logo, my suggestion would be to shift the parts that a human might want to review in a diff to the first part of the url (basically, put the logo last since that's something no human can diff anyway)

**Comment by calebcartwright** [comment]
Criss-crossing answers from Chris and I there :smile: , but (a) good to know there's some dialects/favors that are linebreak friendly and (b) that GitHub indeed doesn't support it and that's why I wasn't able to get it to work :+1:
